core/math/math_funcs.h

```diff
@@ -388,7 +388,7 @@ class Math {
 	}
 
 	static _ALWAYS_INLINE_ bool is_zero_approx(double s) {
-		return abs(s) < CMP_EPSILON;
+		return abs(s) < CMP_EPSILON_64;
```

Comment:
I think the safest option is to only use doubles for the epsilon if both of these are true:
* A double is passed into the methods.
* Godot is compiled with real_t as double.
```suggestion
#ifdef REAL_T_IS_DOUBLE
		return abs(s) < CMP_EPSILON_64;
#else
		return abs(s) < CMP_EPSILON_32;
#endif // REAL_T_IS_DOUBLE
```